src/ppl/src/IOPlacer.cpp

```diff
@@ -907,12 +921,26 @@ void IOPlacer::defineSlots()
   int regular_pin_count = static_cast<int>(netlist_io_pins_->getIOPins().size())
                           - top_layer_pins_count_;
   if (regular_pin_count > slots_.size()) {
+    int min_dist = std::numeric_limits<int>::min();
+    for (int layer_idx : ver_layers_) {
+      const int layer_min_dist = core_->getMinDstPinsX()[layer_idx];
+      min_dist = std::max(layer_min_dist, min_dist);
+    }
+    for (int layer_idx : hor_layers_) {
+      const int layer_min_dist = core_->getMinDstPinsY()[layer_idx];
+      min_dist = std::max(layer_min_dist, min_dist);
+    }
+
+    int64_t die_margin = getBlock()->getDieArea().margin();
+    int64_t increase = computeIncrease(min_dist, regular_pin_count, die_margin);
+
     logger_->error(PPL,
                    24,
                    "Number of IO pins ({}) exceeds maximum number of available "
-                   "positions ({}).",
+                   "positions ({}). Increase the die width or height in {}um.",
```

Comment:
be specific here, if possible. Is it width or height?

The user now has to guess

Reply:
increase from ... to .... will be less ambiguous

Reply:
I also think of something like "Increase the die perimeter from {} to {}". Does it make sense, in your opinion?

Reply:
It is more generic, but it gives more room for the user to choose where they want to increase. It can be only width, only height, or divide the increase value between both.

Reply:
i see, so not constrained to an edge here.

Reply:
Yes. Mainly because in this error, there is no room for the pins, independent of where they're constrained to. So the suggested increase value is to ensure that exists enough positions to place the pins in the boundaries.